paimon-common/src/main/java/org/apache/paimon/io/cache/CacheManager.java

```diff
@@ -29,31 +33,63 @@
 /** Cache manager to cache bytes to paged {@link MemorySegment}s. */
 public class CacheManager {
 
+    private static final Logger LOG = LoggerFactory.getLogger(CacheManager.class);
+
     /**
      * Refreshing the cache comes with some costs, so not every time we visit the CacheManager, but
      * every 10 visits, refresh the LRU strategy.
      */
     public static final int REFRESH_COUNT = 10;
 
-    private final Cache cache;
+    private final Cache dataCache;
+    private final Cache indexCache;
 
     private int fileReadCount;
 
+    @VisibleForTesting
     public CacheManager(MemorySize maxMemorySize) {
-        this(Cache.CacheType.GUAVA, maxMemorySize);
+        this(Cache.CacheType.GUAVA, maxMemorySize, 0);
+    }
+
+    public CacheManager(MemorySize dataMaxMemorySize, double highPrioPoolRatio) {
+        this(Cache.CacheType.GUAVA, dataMaxMemorySize, highPrioPoolRatio);
     }
 
-    public CacheManager(Cache.CacheType cacheType, MemorySize maxMemorySize) {
-        this.cache = CacheBuilder.newBuilder(cacheType).maximumWeight(maxMemorySize).build();
+    public CacheManager(
+            Cache.CacheType cacheType, MemorySize maxMemorySize, double highPrioPoolRatio) {
+        Preconditions.checkArgument(
+                highPrioPoolRatio >= 0 && highPrioPoolRatio < 1,
```

Comment:
Do we need to pay attention to the accuracy of floating-point numbers here, which may lead to judgment errors?

Reply:
e, I think it's ok here

Reply:
ok